### ISSUE TITLE
Limit `prefer-ternary` to assignments

### ISSUE THREAD

**Issue by fregante** [sindresorhus/eslint-plugin-unicorn#1633]
### Description

I tried enabling `prefer-ternary` again on a large codebase and I think I found the only place where I actually prefer its result: assignments.

This also works if lines are long as long as they're wrapped

### Fail

```js
if (a) 
	foo = 1
else 
	foo = 2
```

```js
if (a) 
	return 'Nisi temporibus quae vel officia? Rerum error blanditiis'
else 
	return 'pariatur praesentium repellendus culpa neque veritatis, non assumenda fugiat dicta dolorum '
```

### Pass

```js
foo = a ? 1 : 2
```

```js
return a ?
	'Nisi temporibus quae vel officia? Rerum error blanditiis' : 
	'pariatur praesentium repellendus culpa neque veritatis, non assumenda fugiat dicta dolorum '
```

**Comment by fregante** [comment]
For reference, this is what I _don't_ want fixed:
```js
  if (packageUUID) {
    await client.put(`api/bricks/${packageUUID}/`, data);
  } else {
    await client.post("api/bricks/", data);
  }
```

but the rule is currently auto-fixing it to:

```js
  await (packageUUID
    ? client.put(`api/bricks/${packageUUID}/`, data)
    : client.post("api/bricks/", data));
```

**Comment by sindresorhus** [comment]
Yeah, after some experience with the rule, I tend to agree.

**Comment by KurtPreston** [comment]
We've been trying to fully sunset our use of tslint. We've found replacements for nearly all tslint's rules except for **prefer-conditional-expression**:
https://palantir.github.io/tslint/rules/prefer-conditional-expression/

This is very similar to unicorn's **prefer-ternary** rule, except that it is limited to assignments. If this rule supported an option limiting it to assignments, we could remove tslint from our codebase.

**Comment by fregante** [comment]
Then what about `return await`?

I don’t think that granularity is needed, it seems that most people here agree that it should only affect return and assignments. Potentially `yield` falls into the `return` category. 

The logic here would be: the value is being captured. 

As for `await`, I don’t think it ever makes sense, I've never seen this and liked it:

```JS
await (anything inside parens)
```

**Comment by ipanasenko** [comment]
Hey guys. I went ahead and made a PR: https://github.com/sindresorhus/eslint-plugin-unicorn/pull/2856
Would appreciate a review 🙏
